repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy

```diff
@@ -79,20 +75,7 @@ class KeystoneV2SelfValidationTest extends ReposeValveTest {
         mc.getHandlings().get(0).getRequest().getHeaders().getFirstValue("x-tenant-name") == "mytenantname"
 
         and: "The X-Authenticated-By headers should be present at the origin service."
-        def foundPassword = false
-        def foundMultiFactor = false
-        for (Header header : mc.getHandlings().get(0).getRequest().getHeaders()) {
-            if ("X-Authenticated-By".equalsIgnoreCase(header.name)) {
-                if ("Password".equalsIgnoreCase(header.value)) {
-                    foundPassword = true
-                }
-                if ("Mfa".equalsIgnoreCase(header.value)) {
-                    foundMultiFactor = true
-                }
-            }
-        }
-        assertTrue(foundPassword)
-        assertTrue(foundMultiFactor)
+        mc.getHandlings().get(0).getRequest().getHeaders().findAll("X-Authenticated-By").sort() == ["MFA", "PASSWORD"]
```

Comment:
I blame you for the MFA thing. I copypasta'd your line. You knew MFA was going to be replaced!

I'm aware. My problem with accessing fields directly is that access protection (e.g., private) mechanisms are ignored. If I remember correctly, Groovy will not generate an accessor method for restricted fields. So by always using accessor methods, whether they be natural or synthetically produced by Groovy, I am assured that I have access to the field in question. As far as indexing a collection goes, I don't really have a good reason to prefer `get(...)` to `[...]`. The latter is just syntactic sugar, right? I suppose the former could be copypasta'd into Java, and that's something...  Have I ever mentioned that I'm not really a fan of Groovy? 😛